src/main/java/com/microsoft/graph/http/BaseCollectionPage.java

```diff
@@ -66,9 +67,13 @@
      * @param nextRequestBuilder the request builder for the next page
      */
     public BaseCollectionPage(final List<T1> pageContents, final T2 nextRequestBuilder) {
-        // CollectionPages are never directly modifiable, either 'update'/'delete' the specific child or 'add' the new
-        // object to the 'children' of the collection.
-        this.pageContents = Collections.unmodifiableList(pageContents);
+        if(pageContents == null) {
+            this.pageContents = new ArrayList<T1>();
+        } else {
+            // CollectionPages are never directly modifiable, either 'update'/'delete' the specific child or 'add' the new
+            // object to the 'children' of the collection.
+            this.pageContents = Collections.unmodifiableList(pageContents);
+        }
```

Comment:
Q: Any reason why we shouldn't keep the immutable behavior for the edge case?

```suggestion
        // CollectionPages are never directly modifiable, either 'update'/'delete' the specific child or 'add' the new
        // object to the 'children' of the collection.
        this.pageContents = Collections.unmodifiableList(pageContents == null ? new ArrayList<T1>() : pageContents);
```

Reply:
yep, no particular reason, I don't know why I didn't go with this to start with. Thanks!